File: test-suite/src/test/java/io/micronaut/jsonschema/test/ConfigurationErrorsExample.java
package io.micronaut.jsonschema.test;

import io.micronaut.jsonschema.configuration.validator.ConfigurationErrors;
import jakarta.inject.Singleton;

@Singleton
class ConfigurationErrorsExample {

    private final ConfigurationErrors configurationErrors;

    ConfigurationErrorsExample(ConfigurationErrors configurationErrors) {
        this.configurationErrors = configurationErrors;
    }

    // tag::runtime-api[]
    void logConfigurationErrors() {
        configurationErrors.getCurrentErrors().forEach(System.out::println);
    }
    // end::runtime-api[]
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/jsonschema/test/ConfigurationErrorsExample.kt
package io.micronaut.jsonschema.test

import io.micronaut.jsonschema.configuration.validator.ConfigurationErrors
import jakarta.inject.Singleton

@Singleton
class ConfigurationErrorsExample(private val configurationErrors: ConfigurationErrors) {

    // tag::runtime-api[]
    fun logConfigurationErrors() {
        configurationErrors.currentErrors.forEach(::println)
    }
    // end::runtime-api[]
}
